bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -359,8 +379,21 @@ public void runWithFlags(boolean force, boolean suspendMajor, boolean suspendMin
             minorCompacting.set(true);
             doCompactEntryLogs(minorCompactionThreshold);
             lastMinorCompactionTime = System.currentTimeMillis();
+            if (!enableMajorCompaction || suspendMajor) {
```

Comment:
Given the if condition above, we enter this block only if MajorCompaction is enabled right?